lib/dialects/abstract/query-generator.js

```diff
@@ -1688,6 +1688,16 @@ const QueryGenerator = {
         ) {
           subQueryOrder.push(this.quote(order, model, '->'));
         }
+
+        if (subQuery) {
+          // Handle case where sub-query renames attribute we want to order by,
+          // see https://github.com/sequelize/sequelize/issues/8739
+          const subQueryAttribute = options.attributes.find(a => Array.isArray(a) && a[0] === order[0] && a[1]);
+          if (subQueryAttribute) {
+            order[0] = new Utils.Col(subQueryAttribute[1]);
```

Comment:
These wont work when order contain multiple attributes being renamed by subQuery. Please add a test for two or more attributes

Reply:
@sushantdhiman I believe it actually will work since we loop over each `order` clause above. I added some tests which should confirm that. Let me know if I'm mistaken.